lib/modules/dosomething/dosomething_user/dosomething_user.module

```diff
@@ -103,9 +103,36 @@ function dosomething_user_menu() {
     'file' => 'dosomething_user.admin.inc',
     'weight' => 500,
   );
+
+    return $items;
+}
+
+function dosomething_user_menu_alter(&$items) {
+  unset($items['user']);
+  unset($items['user/%user']);
+  $items['user/%'] = array(
+    'page callback' => 'dosomething_user_get_view',
+    'page arguments' => array(1),
+    'access callback' => TRUE,
+  );
+  unset($items['user/%user']);
+
   return $items;
 }
 
+function dosomething_user_get_view($id) {
+  if (is_numeric($id)) {
+    $account = user_load($id);
+    return user_view($account);
+  }
+  else {
+    $phoenix_user = dosomething_user_get_user_by_northstar_id($id);
+    if (!$phoenix_user) {
+      drupal_not_found();
+    }
+    drupal_goto('user/' . $phoenix_user->uid);
```

Comment:
This  `drupal_goto` looks suspect, per issue you described in Slack (the user profile template is wrapped in default account page header)? 

Does `return user_view($phoenix_user)` fix ?  If so, it's minor but the `if/else` could just set our `$account` User object, and then `return user_view($account)` outside of the if/else block